lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -621,6 +621,21 @@ ModuleDiscriminator::ModuleDiscriminator(Operation *op) {
         addType("INPUT_TYPE", op->getOperand(0));
         addType("OUTPUT_TYPE", op->getResult(0));
       })
+      .Case<handshake::SpecCommitOp, handshake::SpecSaveOp>([&](auto) {
+        addType("DATA_TYPE", op->getOperand(0));
+      })
+      .Case<handshake::SpeculatorOp>([&](auto) {
+        addType("DATA_TYPE", op->getOperand(0));
+        addUnsigned("FIFO_DEPTH", 32); // temp
+      })
+      .Case<handshake::SpecSaveCommitOp>([&](auto) {
+        addType("DATA_TYPE", op->getOperand(0));
+        addUnsigned("FIFO_DEPTH", 32); // temp
+      })
+      .Case<handshake::SpeculatingBranchOp>([&](auto) {
+        addType("SPEC_TAG_DATA_TYPE", op->getOperand(0));
```

Comment:
I don't understand the exact requirements of what happens here to well, but why can't this be hard-coded to a 1-bit integer?

Reply:
Here, `SPEC_TAG_DATA_TYPE` is defined by the bit width of the `tagFromOperand`.
It is important not to hard-code this value as a 1-bit channel.
The VHDL entity of speculating branch needs to receive the entire `tagFromOperand` channel, rather than just the `spec_tag` signal, in the current implementation (without bundle and unbundle ops). If the bit width of `tagFromOperand` is actually anything other than 1 bit, a bit-width mismatch error will occur when running the simulation.

Reply:
got it! curious exactly where we should extract which tag is the spec tag, but possibly this is a later question